bindings/python/CMakeLists.txt

```diff
@@ -0,0 +1,99 @@
+# To build python bindings we need a python executable and boost python module. Unfortunately,
+# their names might not be interlinked and we can not implement a general solution.
+# The code below assumes default boost installation, when the module for python 2 is named
+# 'python' and the module for python 3 is named 'python3'.
+# To customize that one can provide the version suffix for the Boost::python module via
+# 'boost-python-suffix' variable when invoking cmake.
+# E.g. on Gentoo with python 3.6 and Boost::python library name 'libboost_python-3.6.so'
+# the parameter would be -Dboost-python-suffix="-3.6" (notice the '-' character).
+
+find_package(PythonInterp REQUIRED)
+find_package(PythonLibs REQUIRED)
+
+if (NOT boost-python-suffix)
+	# use active python
+	if (PYTHON_VERSION_STRING VERSION_GREATER_EQUAL "3")
+		set(_boost_python_suffix "${PYTHON_VERSION_MAJOR}")
+	else()
+		set(_boost_python_suffix "") # to overwrite possible value from a previous run
+	endif()
+else()
+	set(_boost_python_suffix "${boost-python-suffix}")
+endif()
+
+find_package(Boost REQUIRED COMPONENTS python${_boost_python_suffix})
+
+execute_process(COMMAND
+  ${PYTHON_EXECUTABLE} -c "import distutils.sysconfig;
+print(distutils.sysconfig.get_python_lib(plat_specific=True, prefix=''));
+print(distutils.sysconfig.get_config_var('EXT_SUFFIX'))"
```

Comment:
Python 2.7 doesn't define `EXT_SUFFIX` so the output file of `python-libtorrent` ends up being named `libtorrentNone`.

Reply:
Refactored this piece, thanks! I did not test with python 2 :(